OPAL/br/src/main/scala/org/opalj/br/android/AndroidManifestKey.scala

```diff
@@ -0,0 +1,68 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package br
+package android
+
+import scala.xml.Elem
+import scala.xml.Node
+
+import org.opalj.br.ClassFile
+import org.opalj.br.ObjectType
+import org.opalj.br.analyses.ProjectInformationKey
+import org.opalj.br.analyses.ProjectInformationKeys
+import org.opalj.br.analyses.SomeProject
+
+/**
+ * A ProjectInformationKey that is used to parse an AndroidManifest.xml. It returns a map of Android components and
+ * a ListBuffer of all IntentFilters defined in the manifest.
+ * The AndroidManifest.xml can be set as initialization data.
+ *
+ * @author Julius Naeumann
+ */
+object AndroidManifestKey extends ProjectInformationKey[Option[AndroidManifest], Elem] {
+    // Constants for the IntentFilter
+    val ACTION_MAIN = "android.intent.action.MAIN"
+    val CATEGORY_LAUNCHER = "android.intent.category.LAUNCHER"
+
+    override def requirements(project: SomeProject): ProjectInformationKeys = Nil
+
+    override def compute(project: SomeProject): Option[AndroidManifest] = {
+        val manifest = project.getProjectInformationKeyInitializationData(AndroidManifestKey)
```

Comment:
Might want to add the type of this val to better understand the semantics. Also, would it be sensible to not require clients to read in the XML themselves, but provide a path or file instead? Downside would be that supplying a programmatically produced manifest instead of an existing one in a file would be harder.

Reply:
I still think this val could need an explicit type to better understand the logic here

Reply:
done